File: Soal2.py
def dense_ranking(total_players, scores, num_games, gits_scores):
    ranked_scores = list(set(scores))  # Remove duplicates and sort the scores in ascending order
    ranked_scores.sort(reverse=True)

    rankings = []

    for score in gits_scores:
        scores.append(score)
        scores.sort(reverse=True)

        rank = 1
        for i in range(1, len(scores)):
            if scores[i] < scores[i - 1]:
                rank += 1

            if scores[i] == score:
                rankings.append(rank)
                break

    return rankings

def main():
    try:
        total_players = int(input("Masukkan jumlah pemain: "))
        scores = list(map(int, input("Masukkan daftar skor pemain (dipisahkan oleh spasi): ").split()))

        num_games = int(input("Masukkan jumlah permainan yang diikuti oleh GITS: "))
        gits_scores = list(map(int, input("Masukkan skor GITS setelah setiap permainan (dipisahkan oleh spasi): ").split()))

        result = dense_ranking(total_players, scores, num_games, gits_scores)
        print("Sampel Output:")
        print(*result)

    except ValueError:
        print("Masukkan angka yang valid!")

if __name__ == "__main__":
    main()

File: Soal2.kt
fun denseRanking(scores: IntArray, gitsScores: IntArray): IntArray {
    val sortedScores = scores.distinct().sortedDescending()
    val ranksMap = mutableMapOf<Int, Int>()

    var rank = 1
    for (score in sortedScores) {
        ranksMap[score] = rank
        rank++
    }

    val gitsRanks = IntArray(gitsScores.size)
    for (i in gitsScores.indices) {
        val score = gitsScores[i]
        gitsRanks[i] = ranksMap.getOrElse(score) { sortedScores.size + 1 }
    }

    return gitsRanks
}

fun main() {
    print("Masukkan jumlah pemain: ")
    val numPlayers = readLine()?.toIntOrNull()

    if (numPlayers != null && numPlayers > 0) {
        print("Masukkan skor pemain (pisahkan dengan spasi): ")
        val scores = readLine()?.split(" ")?.mapNotNull { it.toIntOrNull() }?.toIntArray()

        if (scores != null && scores.size == numPlayers) {
            print("Masukkan jumlah permainan yang diikuti GITS: ")
            val numGames = readLine()?.toIntOrNull()

            if (numGames != null && numGames > 0) {
                print("Masukkan skor yang didapatkan GITS (pisahkan dengan spasi): ")
                val gitsScores = readLine()?.split(" ")?.mapNotNull { it.toIntOrNull() }?.toIntArray()

                if (gitsScores != null && gitsScores.size == numGames) {
                    val rankings = denseRanking(scores, gitsScores)
                    println("Output: ${rankings.joinToString(" ")}")
                } else {
                    println("Masukkan skor GITS dengan benar (jumlah sesuai dengan yang diminta).")
                }
            } else {
                println("Masukkan jumlah permainan GITS dengan benar (angka positif).")
            }
        } else {
            println("Masukkan skor pemain dengan benar (jumlah sesuai dengan yang diminta).")
        }
    } else {
        println("Masukkan jumlah pemain dengan benar (angka positif).")
    }
}
